spec/sidekiq-instrument/server_middleware_spec.rb

```diff
@@ -223,7 +225,38 @@
           expect(Sidekiq::Instrument::Statter.dogstatsd).not_to receive(:time)
           expect(
             Sidekiq::Instrument::Statter.dogstatsd
-          ).to receive(:increment).with('sidekiq.error', expected_dog_options).once
+          ).to receive(:increment).with('sidekiq.error', expected_error_dog_options).once
+
+          begin
+            MyWorker.perform_async
+          rescue StandardError
+            nil
+          end
+        end
+      end
+
+      context 'when the error is RedisRateLimit::Throttle::LockFailedError' do
+        let(:expected_lock_error_dog_options) { { tags: ['queue:default', 'worker:my_worker', 'error:RedisRateLimit::Throttle::LockFailedError'] } }
+        let(:lock_error) { RuntimeError.new('foo') }
+
+        before do
+          # force the error's class name to be the RedisRateLimit error type string
+          # it's an external error class and we can't actually create an instance of the error
+          allow(lock_error).to receive_message_chain(:class, :name).and_return("RedisRateLimit::Throttle::LockFailedError")
```

Comment:
Isn't there a more generic error this internal error is based on that we could capture?

Reply:
No. In fact I think this gem has no business being external anymore 🤷‍♀️

Reply:
Sorry, which gem are you referring to? sidekiq-instrument or RedisRateLimit?

Reply:
RedisRateLimit is already internal. I think this gem really doesn't make sense to be external, considering:
- All of the metrics before this past week were very inaccurate
- The Redis counts will remain inaccurate and need complete reworking/removal
- Making this gem internal would allow us to reference our other internal gems from it, for example in the case of the error class from RedisRateLimit